### ISSUE TITLE
Resteasy reactive: @CircuitBreaker ignored when using RestClientBuilder

### ISSUE THREAD

**Issue by lthasle** [quarkusio/quarkus#23549]
### Describe the bug

@CircuitBreaker, it is ignored when a RestClient is build with the RestClientBuilder

se also:
https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Resteasy.20reactive.3A.20.40CircuitBreaker.20ignored.20when.20using.20RestCl

### Expected behavior

@CircuitBreaker should be working on clients build by the RestClientBuilder.
The same way as when it is injected with: @RestClient

### Actual behavior

The CircuitBreaker is ignored on the interface

### How to Reproduce?

run the test: ClientCBOpenTest
[rest-client-reactive-quickstart2.zip](https://github.com/quarkusio/quarkus/files/8032911/rest-client-reactive-quickstart2.zip)

### Output of `uname -a` or `ver`

uname -a Linux hty80028.sb1a.sparebank1.no 4.18.0-348.2.1.el8_5.x86_64 #1 SMP Mon Nov 8 13:30:15 EST 2021 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.10" 2021-01-19 OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.10+9) OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.10+9, mixed mode)

### Quarkus version or git rev

2.7.1.Final

**Comment by geoand** [comment]
cc @michalszynkiewicz @Ladicek

**Comment by Ladicek** [comment]
This used to work with classic RESTEasy, because that implements interceptors on its own. RESTEasy Reactive, for obvious reasons, does not, so it doesn't work.

In the original conversation on Zulip, I said that maybe it would be enough to just instantiate the correct `...CDIWrapper` class, but it doesn't seem as easy as I thought it would be. In my opinion, the question is more like: does RESTEasy Reactive also want to include an [ad hoc, informally-specified, bug-ridden, slow] implementation of [half of] the interceptors specification.

**Comment by michalszynkiewicz** [comment]
just a note the same can be easily done by annotating methods that call the client methods

**Comment by Ladicek** [comment]
For the record, SmallRye Fault Tolerance 5.3.0 has programmatic API that could be used for this. I'm working on a PR to Quarkus as we speak :-)

**Comment by michalszynkiewicz** [comment]
Since it's not easy to implement and known solutions to make it work otherwise exist, I think we should close this.
WDYT @Ladicek @lthasle ?

**Comment by lthasle** [comment]
I guess I can live with that and use the workaround: annotating methods that call the client methods.
But I believe it will be expected to work.

**Comment by Ladicek** [comment]
I recently found why classic RESTEasy had to implement interceptors on its own. Their implementation of MP RestClient scans for the RestClient interfaces and for each, registers a synthetic bean (backed by a `java.lang.reflect.Proxy`). Since synthetic beans don't get intercepted, they just implemented interception manually. That implementation also affects instances created by `RestClientBuilder`, which I'm not sure was ever intentional.

Quarkus works differently, it scans for RestClient interfaces and for each, generates a regular managed bean. That obviously gets intercepted normally, so no need to reimplement interception.

The MP RestClient specification is not explicit about whether interception should happen on `RestClientBuilder`-created instances. At the same time, both the Interceptors specification and CDI specification are clear that `@AroundInvoke` interceptors only interpose on business method invocations. Futhermore, the CDI specification pretty clearly specify what is and isn't a business method invocation (and hence when interception applies). Since Quarkus doesn't support `InjectionTarget`, `UnmanagedInstance` and `InterceptionFactory`, the only case when a method invocation is a business method invocation is when you invoke the method on a contextual reference to a bean. Clearly that's not the case for `RestClientBuilder`-created instances.

To sum up, I agree with @michalszynkiewicz that we won't support this and this issue should be closed.

**Comment by michalszynkiewicz** [comment]
Actually, I'm exploring similar ideas in the works on fault tolerant client. So maybe we'll get to support for this or something similar, but that would be a separate extension, in the quarkiverse.

**Comment by Ladicek** [comment]
I assume that will again only be for CDI beans, right?
